### PR TITLE
[Instructor Training Checkout] Add an longitudinal tabular data exercise.

### PR DESCRIPTION
**Pull request by sechilds**
This commit adds an exercise to the first episode dealing with handling
tabular data longitudinally by taking the differences between
inframmation readings.

### CODE REVIEW

File: _episodes/01-numpy.md [hunk 1]
```diff
@@ -1135,3 +1135,73 @@ the graphs will actually be squeezed together more closely.)
...
+>## Change In Inflamation
+>
+>This patient data is _longitudinal_ in the sense that each row represents a
+>series of observations relating to one individual. This means that change
+>inflamation is a meaningful concept.
+>
+>The `numpy.diff()` function takes a NumPy array and returns the 
+>difference along a specified axis.
+>
+>Which axis would it make sense to use this function along?
```
Comment (tbekolay) [inline _episodes/01-numpy.md]: Can you add a blank blockquote line here to separate the question from the solution? I'm not sure if it's necessary for the markdown rendering, but for those reading the markdown it's helpful at least.

File: _episodes/01-numpy.md [hunk 2]
```diff
@@ -1135,3 +1135,73 @@ the graphs will actually be squeezed together more closely.)
...
+> > concept.
+> >
+> > ~~~
+> > numpy.diff(data, axis=1)
+> > ~~~
+> > {: .python}
+> {: .solution}
+>
+>If the shape of an individual data file is `(60, 40)` (60 rows and 40 columns)
+>, what would the shape of the array be after you run the `diff()` function and
```
Comment (tbekolay) [inline _episodes/01-numpy.md]: Could you put the comma on the previous line? Better to break the parenthetical statement, or move that statement to the next line, than to have a space before the comma.

File: _episodes/01-numpy.md [hunk 3]
```diff
@@ -1135,3 +1135,73 @@ the graphs will actually be squeezed together more closely.)
...
+>why?
```
Comment (tbekolay) [inline _episodes/01-numpy.md]: Empty blockquote after this line here too please!

File: _episodes/01-numpy.md [hunk 4]
```diff
@@ -1135,3 +1135,73 @@ the graphs will actually be squeezed together more closely.)
...
+> > ## Solution
+> > The shape will be `(60, 39)` because there is one fewer difference between
+> > columns than there are columns in the data.
+> {: .solution}
+>
+>How would you find the largest change in inflammation for each patient? Does
+>it matter if the change in inflammation is an increase or a decrease?
```
Comment (tbekolay) [inline _episodes/01-numpy.md]: Here as well.

File: _episodes/01-numpy.md [hunk 5]
```diff
@@ -1135,3 +1135,73 @@ the graphs will actually be squeezed together more closely.)
...
+> > ## Solution
+> > By using the `max()` function after you apply the `diff()` function, you
+> > will get the largest difference between days.
+> > ~~~
+> > numpy.diff(data, axis=1).max(axis=1)
```
Comment (tbekolay) [inline _episodes/01-numpy.md]: We recently made the change to only use NumPy functions and not methods. Could you change this to `numpy.max(numpy.diff(data, axis=1), axis=1)`?

File: _episodes/01-numpy.md [hunk 6]
```diff
@@ -1135,3 +1135,73 @@ the graphs will actually be squeezed together more closely.)
...
+> > The shape will be `(60, 39)` because there is one fewer difference between
+> > columns than there are columns in the data.
+> {: .solution}
+>
+>How would you find the largest change in inflammation for each patient? Does
+>it matter if the change in inflammation is an increase or a decrease?
+> > ## Solution
+> > By using the `max()` function after you apply the `diff()` function, you
+> > will get the largest difference between days.
+> > ~~~
```
Comment (tbekolay) [inline _episodes/01-numpy.md]: I'm not sure if this is required by the Markdown parser either, but for clarity could add newlines before and after code blocks too? I.e.,

```
> > ... end of paragraph.
> > 
> > ~~~
> > numpy.diff(...)
> > ~~~
> > {: .python}
> >
> > Start of next paragraph...
```

File: _episodes/01-numpy.md [hunk 7]
```diff
@@ -1135,3 +1135,73 @@ the graphs will actually be squeezed together more closely.)
...
+> > {: .python}
+> > If a difference is a *decrease*, then the difference will be negative. If
+> > you are interested in the **magnitude** of the change and not just the
+> > direction, the `numpy.absolute()` function will provide that.
+> >
+> > Notice the difference if you get the largest _absolute_ difference
+> > between readings.
+> > ~~~
+> > numpy.absolute(numpy.diff(data, axis=1)).max(axis=1)
```
Comment (tbekolay) [inline _episodes/01-numpy.md]: Here also can you use the `numpy.max` function rather than the `max` method?

**Comment by sechilds** [comment]
Once I thought of this idea it was too much fun not to figure out how to implement. I have made the requested changes and pushed them to this pull request.

**Comment by tbekolay** [comment]
Thanks @sechilds!

**Pull request merged**